File: Constructorio_Net/src/models/Redirect.java
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

/**
 * Constructor.io Redirect ... uses Gson/Reflection to load data in
 */
namespace Constructorio_NET
{
    public class Redirect {

        [JsonPropertyName("data")]
        public RedirectData data;

        [JsonPropertyName("matched_terms")]
        public List<String> matchedTerms;

        [JsonPropertyName("matched_user_segments")]
        public List<String> matchedUserSegments;

    }
}

File: src/Constructorio_NET/models/Redirect.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

/**
 * Constructor.io Redirect
 */
namespace Constructorio_NET.Models
{
    public class Redirect {

        [JsonProperty("data")]
        public RedirectData Data { get; set; }

        [JsonProperty("matched_terms")]
        public List<String> MatchedTerms { get; set; }

        [JsonProperty("matched_user_segments")]
        public List<String> MatchedUserSegments { get; set; }

    }
}